service/statistics/AnalyticsEvents.js

```diff
@@ -0,0 +1,447 @@
+ * getUserMedia.failed.video.720.track.no_data_from_source
+ * getUserMedia.failed.video.undefined.TypeError
+ * getUserMedia.failed.video.undefined.gum.general
+ * getUserMedia.failed.video.undefined.track.no_data_from_source
+ *
+ * TODO: reformat
+ */
+export const GET_USER_MEDIA_FAILED_ = 'getUserMedia.failed';
+
+/**
+ * Properties: value
+ *
+ * Known full event names:
+ * getUserMedia.success
+ * getUserMedia.success.audio
+ * getUserMedia.success.audio.video.180
+ * getUserMedia.success.audio.video.300
+ * getUserMedia.success.audio.video.360
+ * getUserMedia.success.audio.video.720
+ * getUserMedia.success.audio.video.960
+ * getUserMedia.success.audio.video.undefined
+ * getUserMedia.success.desktop
+ * getUserMedia.success.video.180
+ * getUserMedia.success.video.360
+ * getUserMedia.success.video.720
+ * getUserMedia.success.video.960
+ * getUserMedia.success.video.undefined
+ *
+ * TODO: document, reformat
+ */
+export const GET_USER_MEDIA_SUCCESS_ = 'getUserMedia.success';
+
+/**
+ * Properties: none
+ *
+ * Known full event names:
+ * getUserMedia.userCancel.extensionInstall
+ *
+ * TODO: document, reformat
+ */
+export const GET_USER_MEDIA_USER_CANCEL_ = 'getUserMedia.userCancel';
+
+/**
+ * Properties: value
+ *
+ * The "value" property contains the difference in milliseconds between
+ * the ICE establishment time for the P2P and JVB connections (e.g. a value
+ * of 10 would indicate that the P2P was 10ms slower than JVB).
+ */
+export const ICE_ESTABLISHMENT_DURATION_DIFF
+    = 'ice.establishmentDurationDiff';
+
+/**
+ * Properties: none
+ *
+ * TODO: document
+ * TODO: do we need this in addition to _ICE_CONNECTION_STATE?
+ */
+export const ICE_FAILED = 'connection.ice_failed';
+
+/**
+ * Properties: none
+ *
+ * TODO: document
+ */
+export const P2P_ESTABLISHED = 'p2p.established';
+
+/**
+ * Properties: none
+ *
+ * TODO: document
+ */
+export const P2P_FAILED = 'p2p.failed';
+
+/**
+ * Properties: none
+ *
+ * TODO: document
+ */
+export const P2P_SWITCH_TO_JVB = 'p2p.switch_to_jvb';
+
+/**
+ * Properties: none
+ *
+ * TODO: document
+ */
+export const REMOTELY_MUTED = 'conference.remotelyMuted';
+
+/**
+ * Properties: value
+ *
+ * TODO: document
+ *
+ * The "value" property contains the delay in milliseconds between joining
+ * the MUC and receiving a Jingle session-initiate from Jicofo (but not
+ * P2P).
+ */
+export const SESSION_INITIATE = 'session.initiate';
+
+/**
+ * Properties: value
+ *
+ * TODO: document
+ */
+export const SESSION_INITIATE_RECEIVED = 'xmpp.session-initiate';
+
+/**
+ * Properties: none
+ *
+ * TODO: document
+ */
+export const SESSION_TERMINATE = 'session.terminate';
+
+/**
+ * Properties: none
+ *
+ * TODO: document
+ */
+export const SESSION_RESTART = 'session.restart';
+
+/**
+ * Properties: value
+ *
+ * TODO: document
+ */
+export const TRANSPORT_REPLACE_START = 'xmpp.transport-replace.start';
+
+/**
+ * Properties: value
+ *
+ * TODO: document
```

Comment:
When do you suppose these TODOs will get done? Can you feel me imagining myself staring you down, trying to guilt you into doing them right now?

Reply:
I feel zero guilt here :)
It's absolutely impractical for someone to dig and document all these events as part of a cleanup task. I did document the events that I knew about, and that I was working on. I think that's the healthy way to do it: if you work on an event (and therefore know what it does), spend 2 minutes to add a description. I added the TODOs hoping that they will remind people to do this in the future.